hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetrics.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.hadoop.metrics2.lib.MutableRate;
+import org.apache.hadoop.ozone.OzoneConfigKeys;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class which maintains metrics related to using GRPC.
+ */
+@Metrics(about = "GRPC Metrics", context = OzoneConsts.OZONE)
+public class GrpcMetrics {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(GrpcMetrics.class);
+  private static final String SOURCE_NAME =
+      GrpcMetrics.class.getSimpleName();
+
+  private final MetricsRegistry registry;
+  private final boolean grpcQuantileEnable;
+
+  public GrpcMetrics(Configuration conf) {
+    registry = new MetricsRegistry("grpc");
+    int[] intervals = conf.getInts(
+        OzoneConfigKeys.OZONE_GPRC_METRICS_PERCENTILES_INTERVALS_KEY);
+    grpcQuantileEnable = (intervals.length > 0);
+    if (grpcQuantileEnable) {
+      grpcQueueTimeMillisQuantiles =
+          new MutableQuantiles[intervals.length];
+      grpcProcessingTimeMillisQuantiles =
+          new MutableQuantiles[intervals.length];
+      for (int i = 0; i < intervals.length; i++) {
+        int interval = intervals[i];
+        grpcProcessingTimeMillisQuantiles[i] = registry
+            .newQuantiles("grpcQueueTime" + interval
+                    + "s", "grpc queue time in milli second", "ops",
+                "latency", interval);
+        grpcProcessingTimeMillisQuantiles[i] = registry.newQuantiles(
+            "grpcProcessingTime" + interval + "s",
+            "grpc processing time in milli second",
+            "ops", "latency", interval);
+      }
+    }
+    LOG.debug("Initialized " + registry);
+  }
+
+  /**
+   * Create and return GrpcMetrics instance.
+   * @param conf
+   * @return GrpcMetrics
+   */
+  public static synchronized GrpcMetrics create(Configuration conf) {
+    GrpcMetrics metrics = new GrpcMetrics(conf);
+    return DefaultMetricsSystem.instance().register(SOURCE_NAME,
+        "Metrics for using gRPC", metrics);
+  }
+
+  /**
+   * Unregister the metrics instance.
+   */
+  public void unRegister() {
+    DefaultMetricsSystem.instance().unregisterSource(SOURCE_NAME);
+  }
+
+  @Metric("Number of sent bytes")
+  private MutableGaugeLong sentBytes;
+
+  @Metric("Number of received bytes")
+  private MutableGaugeLong receivedBytes;
+
+  @Metric("Queue time")
+  private MutableRate grpcQueueTime;
+
+  // There should be no getter method to avoid
+  // exposing internal representation. FindBugs error raised.
+  private MutableQuantiles[] grpcQueueTimeMillisQuantiles;
+
+  @Metric("Processsing time")
+  private MutableRate grpcProcessingTime;
+
+  // There should be no getter method to avoid
+  // exposing internal representation. FindBugs error raised.
+  private MutableQuantiles[] grpcProcessingTimeMillisQuantiles;
+
+  @Metric("Number of active clients connected")
+  private MutableGaugeInt numOpenClientConnections;
+
+  public void setSentBytes(long byteCount) {
```

Comment:
No, because the counter get initialized to 0 in the constructor [here](https://github.com/apache/ozone/blob/72e1748d28aa813742e1a10a499aecf3ef4c6529/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetricsServerResponseInterceptor.java#L45) and it's value gets updated for every message. The constructor that sets it to 0 get called only once [here](https://github.com/apache/ozone/blob/72e1748d28aa813742e1a10a499aecf3ef4c6529/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/GrpcOzoneManagerServer.java#L105).

Reply:
Now that I'm thinking it, if we replace the gauge with a counter we can avoid keeping track of the local variables in the interceptor classes which will return to 0 if there is a server restart.